### PR TITLE
Changed ov::PartialShape to ov::Shape

### PR DESCRIPTION
**Pull request by ilyachur**
Renamed:
- ngraph::PartialShape -> ov::Shape
- ngraph::Shape -> ov::StaticShape

### CODE REVIEW

**Review by mitruska — COMMENTED** [review]
Okay, `ov::Shape` means old `PartialShape`, `ngraph::PartialShape` (means `ov::Shape`) and `ngraph::Shape` stays as is and is still available. 
So we have to be careful because `Shape` can support dynamism depends on namespace.

**Comment by ilyachur** [comment]
> Okay, `ov::Shape` means old `PartialShape`, `ngraph::PartialShape` (means `ov::Shape`) and `ngraph::Shape` stays as is and is still available.
> So we have to be careful because `Shape` can support dynamism depends on namespace.

Yes it is a main disadvantage of such approach. `ngraph::Shape` is a legacy API.

**Review by tomdol — APPROVED** [review]

**Review by ggalieroc — APPROVED** [review]

**Review by mbencer — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]
https://github.com/openvinotoolkit/openvino/blob/b3428b5312a2a6e2519cd5e43a3e267d175894e2/ngraph/python/src/pyngraph/partial_shape.cpp#L22

I think we should change the doc string of `PartialShape` to reflect changes in API. Maybe leave a note about deprecation or just say it wraps `Shape` in new API?

Other changes are looking fine to me 🚀

**Comment by ilyachur** [comment]
> https://github.com/openvinotoolkit/openvino/blob/b3428b5312a2a6e2519cd5e43a3e267d175894e2/ngraph/python/src/pyngraph/partial_shape.cpp#L22
> 
> I think we should change the doc string of `PartialShape` to reflect changes in API. Maybe leave a note about deprecation or just say it wraps `Shape` in new API?
> 
> Other changes are looking fine to me 🚀

Good catch. @jiwaszki in any case we need to think, how we can save the backward compatibility? I mean from the python API perspective, can we have ngraph::PartialShape for old API and ov::Shape in new one?

**Comment by jiwaszki** [comment]
@ilyachur one of the plan was to copy an old API for Python and freeze it as it is. This would preserve all of the backward compatibility code. 

However as I look at the changes there is a problem of using `PartialShape` as alias for `Shape`... that means if some change is going to appear in C++ in the future, it will affect both Python APIs - old and new at the same time, since C++ core is changing. We need to follow up on this one if we want to keep old Python API frozen. Probably this will require copying core files as well, at least files like `ngraph/core/src/partial_shape.cpp` to keep "frozen state" of the old API as complete as possible.

**Review by tomdol — APPROVED** [review]

**Review by mitruska — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

File: ngraph/core/include/openvino/core/descriptor/input.hpp [hunk 1]
```diff
@@ -95,7 +95,7 @@ class OPENVINO_API Input {
     const ngraph::Shape& get_shape() const;
 
     /// \return the partial shape of the connected output
-    const PartialShape& get_partial_shape() const;
+    const Shape& get_partial_shape() const;
```
Comment (ilya-lavrenov) [inline ngraph/core/include/openvino/core/descriptor/input.hpp:98]: what will we do with such names? Do we have a ticket for it?

Can we depending on whether we have new / old API have different names using `ifdef`?

File: ngraph/core/include/openvino/core/static_shape.hpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+#include <vector>
+
+#include "openvino/core/attribute_adapter.hpp"
+#include "openvino/core/axis_set.hpp"
+#include "openvino/core/core_visibility.hpp"
+#include "openvino/core/strides.hpp"
+
+namespace ov {
+/// \brief StaticShape for a tensor.
+class StaticShape : public std::vector<size_t> {
```
Comment (ilya-lavrenov) [inline ngraph/core/include/openvino/core/static_shape.hpp:18]: can we replace to `std::vector<int64_t>` while old good `ngraph::Shape` has `std::size_t`?

File: ngraph/core/include/openvino/core/descriptor/input.hpp (hunk 1 above)
Comment (ilyachur) [reply ngraph/core/include/openvino/core/descriptor/input.hpp:98]: Yes, we have a ticket for it.

File: ngraph/core/include/openvino/core/static_shape.hpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/core/include/openvino/core/static_shape.hpp:18]: I tried to do it, but a lot of structures uses ngraph::Shape and it doesn't look backward compatible. I propose to return to the question with the `get_partial_shape()` solution

**Comment by ilyachur** [comment]
@ilya-lavrenov Please take a look

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**